[Concurrency] `multiple definition of 'swift::threading::fatal'` after new Threading library

**Describe the bug**
`multiple definition of 'swift::threading::fatal(char const*, ...)'` linking failure happens since https://github.com/apple/swift/pull/59287

**To Reproduce**

Follow the below steps.

```
$ cat main.swift
import Dispatch

@main
struct Main {
  static func main() async {}
}

$ /home/katei/.ghq/github.com/kylef/swiftenv/versions/DEVELOPMENT-SNAPSHOT-2022-06-13-a/usr/bin/swiftc -parse-as-library -static-stdlib
error: link command failed with exit code 1 (use -v to see invocation)
/usr/bin/ld.gold: error: /home/katei/.ghq/github.com/kylef/swiftenv/versions/DEVELOPMENT-SNAPSHOT-2022-06-13-a/usr/lib/swift_static/linux/libswiftCore.a(Errors.cpp.o): multiple definition of 'swift::threading::fatal(char const*, ...)'
/usr/bin/ld.gold: /home/katei/.ghq/github.com/kylef/swiftenv/versions/DEVELOPMENT-SNAPSHOT-2022-06-13-a/usr/lib/swift_static/linux/libswift_Concurrency.a(ThreadingError.cpp.o): previous definition here
```

**Expected behavior**

No link failure


**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.4 LTS
 - Swift Version: `DEVELOPMENT-SNAPSHOT-2022-06-13-a`

--- a/stdlib/public/runtime/CMakeLists.txt
+++ b/stdlib/public/runtime/CMakeLists.txt
@@ -67,6 +67,7 @@ set(swift_runtime_sources
     RuntimeInvocationsTracking.cpp
     SwiftDtoa.cpp
     SwiftTLSContext.cpp
+    ThreadingError.cpp
     AccessibleFunction.cpp)
 
 # Acknowledge that the following sources are known.

--- a/stdlib/public/runtime/Errors.cpp
+++ b/stdlib/public/runtime/Errors.cpp
@@ -465,13 +465,3 @@ void swift::swift_abortDisabledUnicodeSupport() {
                     "Unicode normalization data is disabled on this platform");
 
 }
-
-/// Halt because of a problem in the threading library
-SWIFT_ATTRIBUTE_NORETURN
-SWIFT_FORMAT(1, 2)
-void swift::threading::fatal(const char *msg, ...) {
-  va_list val;
-  va_start(val, msg);
-
-  swift::fatalErrorv(0, msg, val);
-}

--- a/stdlib/public/runtime/ThreadingError.cpp
+++ b/stdlib/public/runtime/ThreadingError.cpp
@@ -1,0 +1,25 @@
+//===--- ThreadingError.cpp - Error handling support code -----------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Debug.h"
+#include "swift/Threading/Errors.h"
+#include <cstdio>
+
+// Handle fatal errors from the threading library
+SWIFT_ATTRIBUTE_NORETURN
+SWIFT_FORMAT(1, 2)
+void swift::threading::fatal(const char *format, ...) {
+  va_list val;
+
+  va_start(val, format);
+  swift::fatalErrorv(0, format, val);
+}

